janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -0,0 +1,221 @@
+            } else if (!Objects.equals(values.get("timed_out"), false)) {
+                throw new IOException("Elasticsearch timeout waiting for yellow status");
+            }
+        }
+    }
+
+    @Override
+    public boolean indexExists(String indexName) throws IOException {
+        boolean exists = false;
+        try {
+            delegate.performRequest("GET", "/" + indexName);
+            exists = true;
+        } catch (IOException e) {
+            if (!e.getMessage().contains("404 Not Found")) {
+                throw e;
+            }
+        }
+        return exists;
+    }
+
+    @Override
+    public void createIndex(String indexName, Settings settings) throws IOException {
+        performRequest("PUT", "/" + indexName, mapper.writeValueAsBytes(settings.getAsMap()));
+    }
+
+    @Override
+    public Map getIndexSettings(String indexName) throws IOException {
+        Response response = performRequest("GET", "/" + indexName + "/_settings", null);
+        try (final InputStream inputStream = response.getEntity().getContent()) {
+            Map<String,RestIndexSettings> settings = mapper.readValue(inputStream, new TypeReference<Map<String, RestIndexSettings>>() {});
+            return settings.get(indexName).getSettings().getMap();
+        }
+    }
+
+    @Override
+    public void createMapping(String indexName, String typeName, XContentBuilder mapping) throws IOException {
+        byte[] bytes = mapping.bytes().toBytes();
+        performRequest("PUT", "/" + indexName + "/_mapping/" + typeName, bytes);
+    }
+
+    @Override
+    public void deleteIndex(String indexName) throws IOException {
+        try {
+            performRequest("DELETE", "/" + indexName, null);
+        } catch (IOException e) {
+            if (!e.getMessage().contains("no such index")) {
+                throw e;
+            }
+        }
+    }
+
+    @Override
+    public void bulkRequest(List<ElasticSearchMutation> requests) throws IOException {
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        for (final ElasticSearchMutation request : requests) {
+            Map actionData = ImmutableMap.of(request.getRequestType().name().toLowerCase(),
+                ImmutableMap.of("_index", request.getIndex(), "_type", request.getType(), "_id", request.getId()));
+            outputStream.write(mapper.writeValueAsBytes(actionData));
+            outputStream.write("\n".getBytes());
+            if (request.getSource() != null) {
+                outputStream.write(mapper.writeValueAsBytes(request.getSource()));
+                outputStream.write("\n".getBytes());
+            }
+        }
+
+        // TODO: Refresh needed for tests but would prefer this in test classes/configuration instead
+        final Response response = performRequest("POST", "/_bulk?refresh=wait_for", outputStream.toByteArray());
```

Comment:
Thanks @krlohnes that's helpful. I'd prefer not to move to that just for this specific issue though I could see needing to revisit if there are broader issues with the current full ES test instance approach (e.g. if the underlying Maven artifacts become unreliable or there are port conflict issues running tests in broader dev/CI environments).